### PR TITLE
Add Translation support

### PR DESCRIPTION
**Pull request by rtm516**
Closes: https://github.com/GeyserMC/Geyser/issues/489

TODO:
- [x] Add loading for translation strings
- [x] Move language files to a separate repo as part of a submodule
- Translate
  - [x] Convert all non-debug logging (regex: `getLogger\(\)\.(info|error|warning|severe|log)\(`)
  - [x] ~Error messages (regex: `throw new [A-Za-z]*\("`)~
  - [x] Commands (needs per player translation adding)
  - [x] Login form
  - [x] Core geyser logging
  - [x] Login/auth system

### CODE REVIEW

File: connector/src/main/java/org/geysermc/connector/utils/TranslationUtils.java [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+
+import org.geysermc.connector.GeyserConnector;
+
+import java.io.InputStream;
+import java.text.MessageFormat;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+
+public class TranslationUtils {
```
Comment (RednedEpic) [inline connector/src/main/java/org/geysermc/connector/utils/TranslationUtils.java:37]: I feel like this should be renamed to `LocaleUtils` as the words `translation` or `translator` are commonly used throughout the codebase in terms of translating packets.

Comment (rtm516) [reply connector/src/main/java/org/geysermc/connector/utils/TranslationUtils.java:37]: We already have a class called LocaleUtils and I didn't want to combine them, I can but I feel like the function names will get confusing then.

Comment (rtm516) [reply connector/src/main/java/org/geysermc/connector/utils/TranslationUtils.java:37]: Could be called `LanguageUtils`?

Comment (RednedEpic) [reply connector/src/main/java/org/geysermc/connector/utils/TranslationUtils.java]: Yes that works.

File: connector/src/main/resources/languages/en_us.properties [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+geyser.language.missing_file=Missing Geyser language file: {0}
+geyser.language.load_failed=Unable to load Geyser locale map for {0}
+
+geyser.locale.fail.latest_version=Unable to get latest Minecraft version
+geyser.locale.fail.asset_cache=Failed to load locale asset cache: {0}
+geyser.locale.fail.invalid=Invalid locale requested to download and load: {0}
+geyser.locale.fail.file=Unable to load locale: {0} ({1})
+geyser.locale.fail.json=Unable to load Java edition locale map for {0}
+geyser.locale.fail.missing=Missing MC locale file: {0}
+geyser.locale.fail.en_us=Unable to download and extract en_us locale!
```
Comment (DoctorMacc) [inline connector/src/main/resources/languages/en_us.properties:54]: Probably makes more sense for this to be `geyser.locale.en_us.download`.

Comment (rtm516) [reply connector/src/main/resources/languages/en_us.properties]: oops just seen this, is it worth making the change now?

**Comment by SinanGentoo** [comment]
Umm...

The language I chose is zh_CN

File: connector/src/main/java/org/geysermc/connector/utils/LanguageUtils.java [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+
+    /**
+     * Get a formatted language string with the given locale for Geyser
+     *
+     * @param key Language string to translate
+     * @param locale Locale to translate to
+     * @param values Values to put into the string
+     * @return Translated string or the original message if it was not found in the given locale
+     */
+    public static String getLocaleStringPly(String key, String locale, Object... values) {
```
Comment (DoctorMacc) [inline connector/src/main/java/org/geysermc/connector/utils/LanguageUtils.java:104]: You mentioned that `Ply` is used in Garry's Mod, but for Minecraft it's probably better to change this to `getLocaleStringPlayer`.

**Comment by DoctorMacc** [comment]
@SinanGentoo re-test when you have a moment?

**Comment by SinanGentoo** [comment]
> @SinanGentoo re-test when you have a moment?

Yeah,I think he works perfectly on the latest pr

**Pull request merged**